File: app/src/main/java/com/jnjcomu/edison/api/APIBuilder.java
package com.jnjcomu.edison.api;

import android.support.annotation.NonNull;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * @author CodeRi13 <ruto1924@gmail.com>
 * @since 2017-04-14
 */

public class APIBuilder {
    private static final String SERVER_ENDPOINT_PROTOCOL = "http";
    private static final String SERVER_ENDPOINT_PORT = "80";
    private static final String SERVER_ENDPOINT_URL = "dev.jnj.club";
    private static final String SERVER_ENDPOINT =
            SERVER_ENDPOINT_PROTOCOL + "://" + SERVER_ENDPOINT_URL + ":" + SERVER_ENDPOINT_PORT;

    @NonNull
    public static EdisonAPI getAPI() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(SERVER_ENDPOINT)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        return retrofit.create(EdisonAPI.class);
    }
}

File: app/src/main/java/com/jnjcomu/edison/api/APIBuilder.kt
package com.jnjcomu.edison.api

import retrofit2.Retrofit

/**
 * @author CodeRi13 <ruto1924@gmail.com>
 * @since 2017-04-14
 */

object APIScheme {
    val SERVER_ENDPOINT_PROTOCOL = "http"
    val SERVER_ENDPOINT_PORT = "80"
    val SERVER_ENDPOINT_URL = "dev.jnj.club"
    val SERVER_ENDPOINT = "$SERVER_ENDPOINT_PROTOCOL://$SERVER_ENDPOINT_URL:$SERVER_ENDPOINT_PORT"
}

val EdisonAPI: EdisonAPISpec = Retrofit.Builder()
        .baseUrl(APIScheme.SERVER_ENDPOINT)
        .build()
        .create(EdisonAPISpec::class.java)